src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,152 @@
+        System.out.println("    OK, I've marked this task as not done yet:");
+        System.out.println(list.get(indexToUnmark).getLabel());
+        System.out.println("____________________________________________________________");
+    }
+
+    public static void handleExitProgram() {
+        System.out.println("____________________________________________________________");
+        System.out.println("    Bye. Hope to see you again soon!");
+        System.out.println("____________________________________________________________");
+    }
+
+    public static void handleExceptions(Exception e) {
+        switch (e.getClass().getName()) {
+            case "duke.IllegalCommandException":
+                IllegalCommandException commandError = (IllegalCommandException) e;
+                if (commandError.type == IllegalCommandExceptionType.COMMAND_DOES_NOT_EXIST) {
+                    System.out.println("____________________________________________________________");
+                    System.out.println("    Error: The command '" + commandError.value + "' does not exist!");
+                    System.out.println("____________________________________________________________");
+                } else if (commandError.type == IllegalCommandExceptionType.MISSING_VALUE) {
+                    System.out.println("____________________________________________________________");
+                    System.out.println("    Error: Missing value after '" + commandError.value + "' command!");
+                    System.out.println("____________________________________________________________");
+                } else if (commandError.type == IllegalCommandExceptionType.MISSING_PARAMETER) {
+                    System.out.println("____________________________________________________________");
+                    System.out.println("    Error: This command requires the '" + commandError.value + "' parameter!");
+                    System.out.println("____________________________________________________________");
+                }
+                break;
+            case "duke.IllegalParameterException":
+                IllegalParameterException parameterError = (IllegalParameterException) e;
+                if (parameterError.type == IllegalParameterExceptionType.PARAMETER_DOES_NOT_EXIST) {
+                    System.out.println("____________________________________________________________");
+                    System.out.println("    Error: The parameter '" + parameterError.value + "' does not exist!");
+                    System.out.println("____________________________________________________________");
+                } else if (parameterError.type == IllegalParameterExceptionType.MISSING_VALUE) {
+                    System.out.println("____________________________________________________________");
+                    System.out.println("    Error: Missing value after '" + parameterError.value + "' parameter!");
```

Comment:
For the error messages, you might want to consider consolidating all of them into named constant under a same file or as enums.